snapcraft/internal/lifecycle.py

```diff
@@ -336,8 +337,16 @@ def _tar_filter(tarinfo):
 
 def containerbuild(step, project_options, output=None, args=[]):
     config = snapcraft.internal.load_config(project_options)
+    remote = os.environ.get('SNAPCRAFT_CONTAINER_BUILDS')
+    # Default remote if it's a truthy value - otherwise it's a remote name
+    try:
+        util.strtobool(remote)
+        remote = None
+    except ValueError:
+        pass
```

Comment:
Can you explain the logic here? If `SNAPCRAFT_CONTAINER_BUILDS` is "0", `remote` is `None`. If `SNAPCRAFT_CONTAINER_BUILDS` is "1", `remote` is `None`. The only time `remote` is not None is if `SNAPCRAFT_CONTAINER_BUILDS` is "foo" or something.

Reply:
This logic could be improved, you also need to warn if the `remote` is valid and if a remote is actually going to be used, bonus for telling people why this remote is being set, something like;

```
Using a remote lxd instance as SNAPCRAFT_CONTAINER_BUILDS is set to use 'shire' as a remote
```

Reply:
splitting out into its own thing might be good